src/gui/programupdater.cpp

```diff
@@ -36,43 +36,39 @@
 #include <QDebug>
 #include <QDesktopServices>
 #include <QRegularExpression>
-#include <QStringList>
 #include <QXmlStreamReader>
 
 #if defined(Q_OS_WIN)
 #include <QSysInfo>
 #endif
 
 #include "base/net/downloadmanager.h"
+#include "base/utils/version.h"
 #include "base/version.h"
 
 namespace
 {
     bool isVersionMoreRecent(const QString &remoteVersion)
     {
-        const QRegularExpressionMatch regVerMatch = QRegularExpression("([0-9.]+)").match(QBT_VERSION);
-        if (regVerMatch.hasMatch())
-        {
-            const QString localVersion = regVerMatch.captured(1);
-            const QVector<QStringRef> remoteParts = remoteVersion.splitRef('.');
-            const QVector<QStringRef> localParts = localVersion.splitRef('.');
+        using Version = Utils::Version<int, 4, 3>;
 
-            for (int i = 0; i < qMin(remoteParts.size(), localParts.size()); ++i)
+        try
+        {
+            const Version newVersion {remoteVersion};
+            const Version currentVersion {QBT_VERSION_MAJOR, QBT_VERSION_MINOR, QBT_VERSION_BUGFIX, QBT_VERSION_BUILD};
+            if (newVersion == currentVersion)
             {
-                if (remoteParts[i].toInt() > localParts[i].toInt())
+                const bool isDevVersion = QString {QLatin1String {QBT_VERSION_STATUS}}.contains(
+                    QRegularExpression {QLatin1String{"(alpha|beta|rc)"}});
+                if (isDevVersion)
                     return true;
```

Comment:
Maybe `return isDevVersion`?

I realize there is possibly a deficiency here. (in old code too). If the remote version is the same dev version as the local one (eg both `4.5.0beta`) it will still think the remote version is newer.

Also, I would like it to be able to differentiate between `4.5.0beta1` and `4.5.0beta2`. Although this is isn't a hard requirement for the PR.

Reply:
>I realize there is possibly a deficiency here. (in old code too). If the remote version is the same dev version as the local one (eg both 4.5.0beta) it will still think the remote version is newer.

I'm not aware that you put testing builds on that channel, is that possible?

>Also, I would like it to be able to differentiate between 4.5.0beta1 and 4.5.0beta2. Although this isn't a hard requirement for the PR.

I would like to defer it.